pravega/src/main/scala/akka/stream/alpakka/pravega/scaladsl/PravegaTable.scala

```diff
@@ -23,61 +24,40 @@ object PravegaTable {
    */
   def source[K, V](scope: String,
                    tableName: String,
-                   keyFamily: String,
-                   tableReaderSettings: TableReaderSettings[K, V]): Source[(K, V), Future[Done]] =
+                   tableReaderSettings: TableReaderSettings[K, V]): Source[TableEntry[V], Future[Done]] =
     Source.fromGraph(
-      new PravegaTableSource[Tuple2[K, V], K, V](te => (te.getKey.getKey, te.getValue),
-                                                 scope,
-                                                 tableName,
-                                                 keyFamily,
-                                                 tableReaderSettings)
+      new PravegaTableSource[K, V](
+        scope,
+        tableName,
+        tableReaderSettings
+      )
     )
 
   /**
-   * Keys, values and key families are extracted from incoming messages and written to Pravega table.
-   * Messages are emitted downstream unchanged.
+   * A flow from key to and Option[value].
    */
-  def writeFlow[K, V](scope: String, tableName: String)(
-      implicit tableWriterSettings: TableWriterSettings[K, V]
-  ): Flow[(K, V), (K, V), NotUsed] =
-    Flow.fromGraph(
-      new PravegaTableWriteFlow[(K, V), K, V]((o: (K, V)) => o, scope, tableName, tableWriterSettings, None)
-    )
+  def readFlow[K, V](scope: String,
+                     tableName: String,
+                     tableSettings: TableSettings[K, V]): Flow[K, Option[V], NotUsed] =
+    Flow.fromGraph(new PravegaTableReadFlow(scope, tableName, tableSettings))
 
   /**
-   * Keys, values and key families are extracted from incoming messages and written to Pravega table.
+   * Keys and values are extracted from incoming messages and written to Pravega table.
    * Messages are emitted downstream unchanged.
    */
-  def writeFlow[K, V](scope: String, tableName: String, familyExtractor: K => String)(
-      implicit tableWriterSettings: TableWriterSettings[K, V]
-  ): Flow[(K, V), (K, V), NotUsed] =
+  def writeFlow[K, V](scope: String,
+                      tableName: String,
+                      tableWriterSettings: TableWriterSettings[K, V]): Flow[(K, V), (K, V), NotUsed] =
```

Comment:
The setting were `implicit` earlier. Is the API better with explicit settings here?
I'm OK with what you believe works best.

Reply:
Yes, that is not very convenient when multiple settings are in the same scope.